relentless/potential/potential.py

```diff
@@ -34,28 +135,135 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
     def save(self, filename):
-        all_params = {}
+        """Saves the data from a coefficient matrix to a file.
+
+        Serializes each parameter value in order to track the internal values
+        of the :py:class:`Variable` object and distinguish it from 'scalar' values.
+
+        Parameters
+        ----------
+        filename : `str`
+            The name of the file to save data in.
+
+        Raises
+        ------
+        TypeError
+            If a value is of an unrecognizable type
+
+        """
+        data = {}
         for key in self:
-            all_params[str(key)] = self.evaluate(key)
+            data[str(key)] = {}
```

Comment:
Since the key will get reused a couple times below, let's save the string version once and recycle it, like
```py
dkey = str(key)
data[dkey] = {}
```